formatter/sarif.go

```diff
@@ -72,7 +72,7 @@ func (l *reviveRunLog) addRules(cfg map[string]lint.RuleConfig) {
 	}
 }
 
-func (l *reviveRunLog) AddResult(failure lint.Failure) {
+func (l *reviveRunLog) addResult(failure lint.Failure) {
```

Comment:
One feedback not directly revlated to this PR, but a linter idea I have by reading this change.

It reminds me unexported-return rule.

Here reviveRunLog is an unexported struct, having unexported methods seems not logic, and I think it's why you made this change

Reply:
I think my idea is stupid if we simply consider a struct like

```go
type fooError struct{}

func (fooError) Error() string {
	return "foo"
}

func whatever() error {
	return fooError{}
}
```

This is valid